go/base/static/js/src/components/views.js

```diff
@@ -36,15 +36,30 @@
       // we call render to output "0 characters used, 0 smses."
       this.render();
     },
+    contains_non_ascii: function(text) {
+      var ascii = true;
+      for (var i = 0; i < text.length; i++) {
+          if (text.charCodeAt(i) > 127) {
+              ascii = false;
+              break;
+          }
+      }
+      return !ascii;
+    },
     render: function() {
       var $p = this.$el.next();
+      var text = this.$el.val();
       if (!$p.hasClass('textarea-char-count')) {
           $p = $('<p class="textarea-char-count"/>');
           this.$el.after($p);
       }
-      this.totalChars = this.$el.val().length;
+      this.containsNonAscii = this.contains_non_ascii(text);
+      this.totalChars = text.length * (1 + this.containsNonAscii);
```

Comment:
It's SMS specific but fairly generic. If there is a non-ascii character the
encoding changes to a 2-byte one and the number of bytes doubles.
On 5 Jun 2015 10:06, "Justin van der Merwe" <notifications@github.com>
wrote:

> In go/base/static/js/src/components/views.js
> <https://github.com/praekelt/vumi-go/pull/1293#discussion_r31794189>:
>
> >        if (!$p.hasClass('textarea-char-count')) {
> >            $p = $('<p class="textarea-char-count"/>');
> >            this.$el.after($p);
> >        }
> > -      this.totalChars = this.$el.val().length;
> > +      this.containsNonAscii = this.contains_non_ascii(text);
> > +      this.totalChars = text.length * (1 + this.containsNonAscii);
>
> Struggling to understand this. Could you explain why we double the total
> chars if it contains ascii characters? Is it a channel-specific thing?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/praekelt/vumi-go/pull/1293/files#r31794189>.
>